src/main/java/fast_reset/client/mixin/MinecraftClientMixin.java

```diff
@@ -0,0 +1,34 @@
+package fast_reset.client.mixin;
+
+import fast_reset.client.FastReset;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.SaveLevelScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.text.TranslatableText;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(MinecraftClient.class)
+public abstract class MinecraftClientMixin {
+
+    @Shadow public abstract void method_29970(Screen screen);
+
+    @Inject(method = "startIntegratedServer(Ljava/lang/String;Lnet/minecraft/util/registry/RegistryTracker$Modifiable;Ljava/util/function/Function;Lcom/mojang/datafixers/util/Function4;ZLnet/minecraft/client/MinecraftClient$WorldLoadAction;)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/MinecraftServer;startServer(Ljava/util/function/Function;)Lnet/minecraft/server/MinecraftServer;"))
+    private void worldWait(CallbackInfo ci) {
+        synchronized (FastReset.saving) {
+            if (!FastReset.saving) {
+                FastReset.LOGGER.info("no save lock active");
+                return;
+            }
+        }
+
+        this.method_29970(new SaveLevelScreen(new TranslatableText("still saving the last world")));
+
+        synchronized(FastReset.saveLock) {
+            FastReset.LOGGER.info("done waiting for save lock");
+        }
+    }

```

Comment:
thoughts on moving this over to MinecraftServer.startServer then checking to see if we have a client and opening the screen?

that way this stuff could still all work server side if someone wants to install the mod on a server.

(this isnt a blocker btw if you say you dont feel like it im still willing to merge things)

Reply:
in its current state its a clientside mod since there is no way to trigger fastreset on a server
i did think about adding a gamerule serverside or smth but yea that would be a thing for another update if i feel like making it, if i (or someone elseI) do that id probably move it but rn it seems there

Reply:
ya 100% non blocking I just wanted to bring it up if you wanted to take it on now but its 100% easy to save for some other time